src/Microsoft.AspNetCore.Server.Kestrel.Https/HttpsConnectionAdapter.cs

```diff
@@ -119,6 +131,32 @@ public Task<IAdaptedConnection> OnConnectionAsync(ConnectionAdapterContext conte
             return new HttpsAdaptedConnection(sslStream);
         }
 
+        private static void ValidateEku(X509Certificate2 certificate)
+        {
+            // For details on enhanced key usage, see https://tools.ietf.org/html/rfc3280#section-4.2.1.13
+            var hasEkuExtension = false;
+            var allowedForServerAuth = false;
+
+            foreach (var extension in certificate.Extensions.OfType<X509EnhancedKeyUsageExtension>())
+            {
+                hasEkuExtension = true;
+                foreach (var oid in extension.EnhancedKeyUsages)
+                {
+                    if (oid.Value.Equals(ServerAuthenticationOid.Value, StringComparison.Ordinal))
+                    {
+                        allowedForServerAuth = true;
+                        goto finish;
```

Comment:
I wish. X509EnhancedKeyUsageExtension.EnhancedKeyUsages is an [OidCollection](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.oidcollection?view=netstandard-2.0), which doesn't implement IEnumerable\<T>.